src/exchanges.js

```diff
@@ -35,10 +21,16 @@ var Publisher = function(conn, channel, entries, exchangePrefix, options) {
   this._channel = channel;
   this._entries = entries;
   this._options = options;
-  if (options.drain) {
-    assert(options.component, "component name for statistics is required");
-    assert(options.process,   "process name for statistics is required");
-    this._reporter = ExchangeReports.reporter(options.drain);
+  if (options.drain || options.component) {
+    console.log('taskcluster-lib-stats is now deprecated!\n' +
+                'Use the `monitor` option rather than `drain`.\n' +
+                '`monitor` should be an instance of taskcluster-lib-monitor.\n' +
+                '`component` is no longer needed. Prefix your `monitor` before use.');
+  }
+
+  var monitor = null;
+  if (options.monitor) {
+    monitor = options.monitor.prefix('pulse-publisher');
```

Comment:
I guess I misunderstood your review in tc-lib-api. Currently we do [this](https://github.com/taskcluster/taskcluster-lib-api/blob/master/src/api.js#L718) in tc-lib-api. When you reviewed it last I just removed the `component` part of it and left the "api" prefix in. I can go remove that there as well and remove this here, or leave them both in since it still kinda makes sense to me. A user of this library is responsible for prefixing it with the name of the component using the publisher, but this can prefix on top of that one. Thoughts?